dask/array/routines.py

```diff
@@ -1843,6 +1843,22 @@ def piecewise(x, condlist, funclist, *args, **kw):
     )
 
 
+@derived_from(np)
+def select(condlist, choicelist, default=0):
+    return blockwise(
+        np.select,
+        "m",
+        condlist,
+        "nm",
+        choicelist,
+        "nm",
```

Comment:
To support arrays with ndims > 1, you can use a tuple that depends on `ndim` something like:
```suggestion
        tuple(range(choicelist[0].ndim + 1)),
```